wrappers/rust/icicle-core/src/msm/tests.rs

```diff
@@ -178,6 +179,172 @@ where
         .unwrap();
 }
 
+pub fn check_msm_batch_shared<C: Curve + MSM<C>>()
+where
+    <C::ScalarField as FieldImpl>::Config: GenerateRandom<C::ScalarField>,
+{
+    // let test_sizes = [1000, 1 << 16]; //TODO - uncomment this line after implementing fast msm
+    let test_sizes = [100];
+    // let batch_sizes = [1, 3, 1 << 4];
+    let batch_sizes = [1, 3]; //TODO - uncomment this line after implementing fast msm
+    let mut stream = IcicleStream::create().unwrap();
+    let precompute_factor = 8;
+    let mut cfg = MSMConfig::default();
+    cfg.stream_handle = *stream;
+    cfg.is_async = true;
+    cfg.ext
+        .set_int(CUDA_MSM_LARGE_BUCKET_FACTOR, 5);
+    cfg.c = 4;
+    runtime::warmup(&stream).unwrap();
+    stream
+        .synchronize()
+        .unwrap();
+    for test_size in test_sizes {
+        // (1) compute MSM with and w/o precompute on main device
+        test_utilities::test_set_main_device();
+        cfg.precompute_factor = precompute_factor;
+        let points = generate_random_affine_points_with_zeroes::<C>(test_size, 10);
+        let mut precomputed_points_d =
+            DeviceVec::<Affine<C>>::device_malloc(cfg.precompute_factor as usize * test_size).unwrap();
+        precompute_bases(HostSlice::from_slice(&points), &cfg, &mut precomputed_points_d).unwrap();
+        for batch_size in batch_sizes {
+            let scalars = <C::ScalarField as FieldImpl>::Config::generate_random(test_size * batch_size);
+            // a version of batched msm without using `cfg.points_size`, requires copying bases
+            // let points_cloned: Vec<Affine<C>> = std::iter::repeat(points.clone())
```

Comment:
remove if not needed

Reply:
This was a confusion in previous versions. Safe to remove @nonam3e